### PR TITLE
update pt-br norme to v4

### PR DESCRIPTION
**Pull request by vitorsantanna2**
I did an update from PT_BR Norme version 3 to PT_BR Norme version 4

### CODE REVIEW

File: pdf/pt_br.norm.tex [hunk 1]
```diff
@@ -157,9 +156,7 @@ \chapter{A norma}
                 arquivos ou diretórios) devem ter os
                 nomes mais explícitos ou mais mnemônicos possíveis.
 
-            \item Usando uma variável global em um projeto onde não é
-                explicitamente permitido é um erro de norma, exceto onde é
-                obrigatório (manuseio de sinais, por exemplo).
+            \item Usando variáveis globais que não são marcadas como constante e estática é proibido e é considerado um erro de norma, há menos que seja explicitamente permitido no projeto.
```
Comment (NiumXp) [inline pdf/pt_br.norm.tex:159]:
```suggestion
            \item Usar variáveis globais é proibido e considerado um erro de norma, há menos que seja explicitamente permitido o uso no projeto.
```

File: pdf/pt_br.norm.tex [hunk 2]
```diff
@@ -463,10 +461,14 @@ \chapter{A norma}
             \item Seus comentários devem estar em inglês. E eles devem ser
                 úteis.
 
-            \item Um comentário não pode ser usado para justificar uma função 
+            \item Um comentário não pode ser usado para justificar declarações ou  uma função 
 				mal feita.
 
         \end{itemize}
+	\warn{
+		Uma declaração ou função mal feita normalmente vem com nomes não explícitos como f1, f2... para as funções e a, b, i, ... para as declarações.
```
Comment (NiumXp) [inline pdf/pt_br.norm.tex:469]:
```suggestion
		Uma declaração ou função mal feita normalmente vem com nomes não inteligíveis como f1, f2, etc. para as funções e, a, b, i, etc. para declarações.
```

File: pdf/pt_br.norm.tex [hunk 3]
```diff
@@ -463,10 +461,14 @@ \chapter{A norma}
             \item Seus comentários devem estar em inglês. E eles devem ser
                 úteis.
 
-            \item Um comentário não pode ser usado para justificar uma função 
+            \item Um comentário não pode ser usado para justificar declarações ou  uma função
```
Comment (NiumXp) [inline pdf/pt_br.norm.tex:464]:
```suggestion
            \item Um comentário não pode ser usado para justificar declarações ou uma função 
```

File: pdf/pt_br.norm.tex [hunk 4]
```diff
@@ -463,10 +461,14 @@ \chapter{A norma}
...
                 úteis.
 
-            \item Um comentário não pode ser usado para justificar uma função 
+            \item Um comentário não pode ser usado para justificar declarações ou  uma função 
 				mal feita.
 
         \end{itemize}
+	\warn{
+		Uma declaração ou função mal feita normalmente vem com nomes não explícitos como f1, f2... para as funções e a, b, i, ... para as declarações.
+		Uma função que tem o único objetivo de evitar a norma, sem um único propósito lógico, também é considerado uma função mal feita.
```
Comment (NiumXp) [inline pdf/pt_br.norm.tex:470]:
```suggestion
		Uma função cujo objetivo é evitar ou burlar a norma, sem um único propósito lógico, também é considerado uma função mal feita.
```

File: pdf/pt_br.norm.tex [hunk 5]
```diff
@@ -463,10 +461,14 @@ \chapter{A norma}
...
+		Por favor lembre se que é desejável ter funções limpas e legíveis que realizem uma clara e simples tarefa cada. Evite qualquer técnica de ofuscação do código como \emph{one-liner}..
```
Comment (NiumXp) [inline pdf/pt_br.norm.tex:471]:
```suggestion
		Por favor, lembre-se de que é desejável ter funções limpas e legíveis que realizem uma clara e simples tarefa. Evite qualquer técnica de ofuscação de código como \emph{one-liner}.
```

File: pdf/pt_br.norm.tex [hunk 6]
```diff
@@ -408,7 +406,7 @@ \chapter{A norma}
             \item Nomes de macro devem ser todos maiúsculos (uppercase).
             \item Você deve recuar caracteres que seguirem \#if, \#ifdef
                 or \#ifndef.
-
+	\item Instruções de pré-processamento são proibidas fora do escopo global.
```
Comment (NiumXp) [inline pdf/pt_br.norm.tex:409]:
```suggestion
            \item Instruções de pré-processamento são proibidas fora do escopo global.
```

File: pdf/pt_br.norm.tex [hunk 7]
```diff
@@ -385,7 +383,7 @@ \chapter{A norma}
 
         \item O header da 42 deve conter informações atualizadas, incluindo o
           criador com login e email, a data de criação e a data da atualização
-          mais recente.
+          mais recente. Cada vez que o arquivo é salvo em disco, a informação deve ser atualizada automaticamente.
```
Comment (NiumXp) [inline pdf/pt_br.norm.tex:386]:
```suggestion
          mais recente. Cada vez que o arquivo for salvo em disco, a informação deve ser atualizada automaticamente.
```

File: pdf/pt_br.norm.tex [hunk 8]
```diff
@@ -236,7 +233,8 @@ \chapter{A norma}
                   Os operadores devem estar no início de uma linha.
 
                 \item Estruturas de controle (if, while ...) devem ter chaves,
-                a menos que contenham uma única linha ou uma única condição.
+                a menos que contenham uma única linha.
+	\item Funções seguidas de chaves, declaradores ou estruturas de controle devem ser precedidas e seguidas de uma nova linha.
```
Comment (NiumXp) [inline pdf/pt_br.norm.tex:237]:
```suggestion
                \item Funções seguidas de chaves, declaradores ou estruturas de controle devem ser precedidas e seguidas de uma nova linha.
```

**Review by NiumXp — CHANGES_REQUESTED** [review]

**Comment by vitorsantanna2** [comment]
Accepted Suggestions

**Review by matthieu42Network — APPROVED** [review]

**Pull request merged**